### PR TITLE
 API micro gateway capability

### PR DESCRIPTION
**Pull request by piotrmsc**
**Description**

Changes proposed in this pull request:

- Added entry to capability README
- Added api-micro-gateway capability description with vision

### CODE REVIEW

File: capabilities/api-gateway.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	* Control outbound traffic for APIs - define a list of external services which the API can access
+	* Control internal API traffic - to specify which services can access APIs internally
+	* Traffic management for different API versions - split traffic between different versions of one API
+	* API failure prevention - enable setting circuit breakers for APIs  
+* Expose GraphQl service apis as REST apis  
+    * Allow applying authorization rules on top
+* Enable developers to create and expose APIs separated on namespace level
+    * Allow exposure of API with Namespace name as a part of domain
+    * Allow blocking communication between services living in different Namespaces
+* Expose services running on different providers - not directly on Kyma
```
Comment (pbochynski) [inline capabilities/api-gateway.md]: Be more specific here and mention proxying requests to services deployed outside kubernetes cluster.

File: capabilities/api-gateway.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+
+
+## Vision
+
+
+* Extend authorization strategies for API's
+	* OAuth2 server issuing access tokens to exposed Kyma APIs (both user and non-user oriented tokens)
+	* OAuth2 proxy securing exposed APIs in Kyma, allowing access based on issued access tokens 
+	* Enable Open Policy Agent policies for authorization and admission control (https://www.openpolicyagent.org/)
```
Comment (tomekpapiernik) [inline capabilities/api-gateway.md]:
```suggestion
	* Enable [Open Policy Agent](https://www.openpolicyagent.org/) policies for authorization and admission control
```

File: capabilities/api-gateway.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	* Support refreshing Oauth2 tokens
```
Comment (tomekpapiernik) [inline capabilities/api-gateway.md]:
```suggestion
	* Support refreshing OAuth2 tokens	
```

File: capabilities/api-gateway.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+* Extend authorization strategies for API's
+	* OAuth2 server issuing access tokens to exposed Kyma APIs (both user and non-user oriented tokens)
+	* OAuth2 proxy securing exposed APIs in Kyma, allowing access based on issued access tokens 
+	* Enable Open Policy Agent policies for authorization and admission control (https://www.openpolicyagent.org/)
+	* Support refreshing Oauth2 tokens	
+	
+* Traffic management for APIs
+	* Control outbound traffic for APIs - define a list of external services which the API can access
+	* Control internal API traffic - to specify which services can access APIs internally
```
Comment (tomekpapiernik) [inline capabilities/api-gateway.md]:
```suggestion
	* Control internal API traffic - specify which services can access APIs internally
```

File: capabilities/api-gateway.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+	* Traffic management for different API versions - split traffic between different versions of one API
+	* API failure prevention - enable setting circuit breakers for APIs
+	  
+* Enable developers to create and expose APIs separated on the namespace level
```
Comment (tomekpapiernik) [inline capabilities/api-gateway.md]:
```suggestion
* Enable developers to create and expose APIs separated on the Namespace level
```

File: capabilities/api-gateway.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    * Allow application to be exposed with Namespace name as a part of the hostname
```
Comment (tomekpapiernik) [inline capabilities/api-gateway.md]:
```suggestion
    * Allow applications to be exposed with Namespace name as a part of their hostname
```

File: capabilities/api-gateway.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    * Allow blocking communication between services living in different Namespaces
+    
+* Enable easy adoption of GraphQl in an API
```
Comment (tomekpapiernik) [inline capabilities/api-gateway.md]:
```suggestion
* Enable easy adoption of GraphQl in APIs 
```

File: capabilities/api-gateway.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+* Enable developers to create and expose APIs separated on the namespace level
+    * Allow application to be exposed with Namespace name as a part of the hostname
+    * Allow blocking communication between services living in different Namespaces
+    
+* Enable easy adoption of GraphQl in an API 
+    * Allow legacy/microservice/serverless applications to be exposed and visible as GraphQl APIs 
+    * Automate configuration and deployment of front end proxies to allow communication using GraphQl
+  
+* Expose services running on different environments
+    * Allow proxying requests to services running outside Kyma - on cloud foundry, other k8s clusters etc.
```
Comment (tomekpapiernik) [inline capabilities/api-gateway.md]:
```suggestion
    * Allow proxying requests to services running outside of Kyma - on Cloud Foundry, other Kubernetes clusters etc.
```

File: capabilities/api-gateway.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    * Allow configuring the same authentication and authorization policies as for services deployed in the Kyma
```
Comment (tomekpapiernik) [inline capabilities/api-gateway.md]:
```suggestion
    * Allow configuring the same authentication and authorization policies as for services deployed in Kyma
```

**Pull request merged**